src/axom/inlet/tests/inlet_restart.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include <iostream>
+
+#include "axom/sidre.hpp"
+
+#include "axom/inlet/Inlet.hpp"
+
+#include "axom/inlet/tests/inlet_test_utils.hpp"
+
+using axom::inlet::Container;
+using axom::inlet::Field;
+using axom::inlet::Inlet;
+using axom::inlet::InletType;
+using axom::inlet::Proxy;
+using axom::sidre::DataStore;
+
+template <typename InletReader>
+Inlet createBasicInlet(DataStore* ds,
+                       const std::string& luaString = {},
+                       bool enableDocs = true,
+                       bool reconstruct = true)
+{
+  std::unique_ptr<InletReader> reader(new InletReader());
+  if(!luaString.empty())
+  {
+    reader->parseString(axom::inlet::detail::fromLuaTo<InletReader>(luaString));
+  }
+
+  return Inlet(std::move(reader), ds->getRoot(), enableDocs, reconstruct);
+}
+
+template <typename InletReader>
+class inlet_restart : public ::testing::Test
+{ };
+
+TYPED_TEST_SUITE(inlet_restart, axom::inlet::detail::ReaderTypes);
+
+TYPED_TEST(inlet_restart, simple_scalars)
+{
+  std::string testString = "foo = true; bar = false";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  // Check for existing fields
+  inlet.addBool("foo", "foo's description");
+  inlet.addBool("bar", "bar's description");
+
+  // No input provided - the datastore should already contain all the data
+  Inlet restartInlet = createBasicInlet<TypeParam>(&ds);
+
+  ASSERT_TRUE(restartInlet.contains("foo"));
+  ASSERT_TRUE(restartInlet.contains("bar"));
+
+  bool value = false;
+  // Check for existing fields
+  value = restartInlet["foo"];
+  EXPECT_TRUE(value);
+
+  value = restartInlet.get<bool>("foo");
+  EXPECT_TRUE(value);
+
+  value = restartInlet["bar"];
+  EXPECT_FALSE(value);
+
+  value = restartInlet.get<bool>("bar");
+  EXPECT_FALSE(value);
+}
+
+TYPED_TEST(inlet_restart, simple_scalars_repeat_schema)
+{
+  std::string testString = "foo = true; bar = false";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  // Check for existing fields
+  inlet.addBool("foo", "foo's description");
+  inlet.addBool("bar", "bar's description");
+
+  // No input provided - the datastore should already contain all the data
+  Inlet restartInlet = createBasicInlet<TypeParam>(&ds);
+
+  // Check for existing fields
+  restartInlet.addBool("foo", "foo's description");
```

Comment:
Suggestion: Consider adding a comment here that this one has `addBool()` calls, while the previous test (`simple_scalars`) didn't ?